### PR TITLE
TDL-17520: Primary Key is not unique for the purchase_log stream

### PR DESCRIPTION
**Pull request by harshpatel4crest**
# Description of change
TDL-17520: Primary Key is not unique for the purchase_log stream
- Used `['date', 'email_hash', 'extid', 'message_id', 'price', 'channel']` as the Primary Key
- Updated the Primary Key in the integration test

**NOTE**: the build is failing due to integration tests failure but that is fixed and all tests are written as part of #5 

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tests/base.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import os
 import unittest
-
+from tap_tester import connections, menagerie, runner
```
Comment (savan-chovatiya) [inline tests/base.py:3]: @harshpatel4crest Please add a PR comment for this line addition here.

Comment (harshpatel4crest) [reply tests/base.py:3]: When running the existing tests, [following error](https://app.circleci.com/pipelines/github/singer-io/tap-sailthru/29/workflows/bb5de5b1-fbf2-4ea8-ba0d-06d5b3f97830/jobs/32?invite=true#step-109-29) was coming. Hence imported **connections, menagerie, runner**

**Review by savan-chovatiya — APPROVED** [review]

**Review by dbshah1212 — APPROVED** [review]

File: tests/base.py [hunk 2]
```diff
@@ -122,8 +122,7 @@ def run_and_verify_check_mode(self, conn_id):
 
     def get_properties(self, original_properties=True):
         properties = {
-            'start_date': '2017-01-01T00:00:00Z',
-            'user_agent': 'tap-sailthru tap-tester@test.com',
```
Comment (RushT007) [inline tests/base.py:126]: Document the reason for removal of user-agent property.

Comment (harshpatel4crest) [reply tests/base.py:126]: @RushT007, when we run tap tester with user_agent, we were [getting error](https://app.circleci.com/pipelines/github/singer-io/tap-sailthru/28/workflows/1df67b33-1b49-4883-8dbc-0341678b1f98/jobs/31?invite=true#step-109-28), which says that **user_agent** is not expected from the config. Hence, Kyle asked to remove user_agent from the base file ([here](https://talend.slack.com/archives/C01TGC4QAUE/p1644239194004059?thread_ts=1643885415.059929&cid=C01TGC4QAUE)).

**Review by RushT007 — APPROVED** [review]

**Review by KrisPersonal — APPROVED** [review]

**Pull request merged**